typescript/infra/src/core/govern.ts

```diff
@@ -77,25 +94,70 @@ export class AbacusCoreGovernor<Chain extends ChainName> {
     });
   }
 
-  async handleValidatorViolation(violation: ValidatorViolation) {
-    const validatorManager = violation.data.validatorManager;
-    switch (violation.data.type) {
-      case ValidatorViolationType.EnrollValidator: {
-        const call = await validatorManager.populateTransaction.enrollValidator(
-          violation.expected,
-        );
-        this.pushCall(violation.chain as Chain, call as types.CallData);
+  // pushes calls which reconcile actual and expected sets on chain
+  protected async pushSetReconcilationCalls<T>(reconcile: {
+    chain: ChainName;
+    actual: Set<T>;
+    expected: Set<T>;
+    add: (elem: T) => Promise<PopulatedTransaction>;
+    remove: (elem: T) => Promise<PopulatedTransaction>;
+  }) {
+    let txs: PopulatedTransaction[] = [];
+    // add expected - actual elements
+    utils
+      .difference(reconcile.expected, reconcile.actual)
+      .forEach(async (item) => txs.push(await reconcile.add(item)));
+    // remove actual - expected elements
+    utils
+      .difference(reconcile.actual, reconcile.expected)
+      .forEach(async (item) => txs.push(await reconcile.remove(item)));
+    // push calls
+    txs.forEach((tx) =>
+      this.pushCall(reconcile.chain as Chain, tx as types.CallData),
+    );
+  }
+
+  async handleAbacusConnectionManagerViolation(
+    violation: AbacusConnectionManagerViolation,
+  ) {
+    const abacusConnectionManager = violation.contract;
+    switch (violation.abacusConnectionManagerType) {
+      case AbacusConnectionManagerViolationType.EnrolledInboxes: {
+        const typedViolation = violation as EnrolledInboxesViolation;
+        const remoteId = ChainNameToDomainId[typedViolation.remote];
+        this.pushSetReconcilationCalls({
```

Comment:
Shouldn't we await here? Should we have a lint rule that catches this?

Reply:
we do have a lint against this
ig its not running because base isn't `main`